### PR TITLE
feat: reflect the session's actual model/effort in the composer

### PR DESCRIPTION
**Pull request by AprilNEA**
## Why

The composer's model/Effort picker showed placeholders (`Default` / `Effort`) instead of the real values. The root cause isn't cosmetic: the picker held **write-only optimistic local state** (initially null), and the data plane **never told the client which model/effort the session is actually running on** — unlike the mode and approval-policy axes, which already have downstream `current-*-update` reflection + engine caching/replay. Model and effort were **upstream-only** in the wire contract.

## What

Mirror the approval-policy reflection end-to-end so the composer shows the true value:

- **schema/wire**: add `model-update` / `effort-update` agent events; **bump `WIRE_PROTOCOL_VERSION` 17 → 18** (Invariant 1 — rebuild/restart daemon + all clients).
- **client-core**: `buildConversation` folds them into `currentModel` / `currentEffort`.
- **engine**: cache both per session and replay on `session.attach` (so a reconnecting client doesn't fall back to a placeholder).
- **agent-adapter**:
  - claude-code reflects the **served model** from the `system:init` frame and every assistant frame (`syncModel`), and emits on `set-model`. Effort has no message-stream carrier, so it learns the **resolved** level (after any per-model downgrade) from a read-only `Stop` hook (`input.effort.level`) — only when the user hasn't explicitly picked one; an explicit pick (incl. `ultracode`/`max`) stays authoritative via `set-effort`.
  - codex reflects model/effort on pick and at start (no live channel for the config default, so a never-set fresh session still shows a placeholder).
- **composer**: drop the local optimistic state; read the server-reflected `currentModel` / `currentEffort`, unifying with how mode / approval-policy already work. Picks are fire-and-forget and reflect back via the session events. Placeholders now appear only before the first turn (genuinely unknown).
- **dev mock**: emits both events so mock mode shows real values too.

## Verification

- Full vitest suite passes (474 passed, 2 skipped) + typecheck across all touched packages.
- New unit tests: `set-model`/`set-effort` emit reflection events; the `init` frame reflects the served model; the `Stop` hook learns the default only when effort is unset.
- ⚠️ Real-app visual verification (observe in the running app) not yet done — mock mode can be used to eyeball the composer showing concrete values.

## Notes

- Wire bump to v18 — check no other in-flight branch also bumps it (parallel bumps merge silently).
- new-session composer's effort picker remains cosmetic (never entered `StartOptions`); only its model is wired to local reflection. Wiring new-session effort is out of scope (needs a `StartOptions.effort`).

### CODE REVIEW

File: packages/agent-adapter/src/native/claude-code.ts [hunk 1]
```diff
@@ -298,6 +299,26 @@ export class ClaudeCodeAdapter extends BaseAgentAdapter {
     this.emitApprovalPolicy(this.approvalPolicyState());
   }
 
+  /** Reflect the served model the CLI reports — the init message and every assistant frame carry it,
+   * so the client shows the true model even when the session started without a requested one. Dedup
+   * lives in `emitModel`. */
+  private syncModel(model: string | undefined): void {
+    if (model) this.emitModel(model);
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply packages/agent-adapter/src/native/claude-code.ts:306]: Good catch — real bug. Fixed in b24bf66: added `resolveModel` in `agent-models.ts` that maps a served snapshot id back to its picker alias by prefix (exact match first, so `gpt-5.4-mini` never resolves to `gpt-5.4`), used for both the trigger label and the radio value. Kept the wire value as the served id and normalized in the UI layer that owns the catalog. Added `agent-models.test.ts` covering the Haiku snapshot case and the prefix edge case.

File: packages/agent-adapter/src/native/claude-code.ts [hunk 2]
```diff
@@ -298,6 +299,26 @@ export class ClaudeCodeAdapter extends BaseAgentAdapter {
...
+  }
+
+  /** Read-only `Stop` hook that learns the CLI's *resolved* effort (after any per-model downgrade)
+   * so a session the user never set an effort on still reflects a real value instead of a
+   * placeholder. Skipped once the user picks explicitly: that pick — including `ultracode`/`max`,
+   * which this hook's base-level field can't express — is authoritative and emitted by `onSetEffort`.
+   * The `effort` field is absent on models without effort support, in which case nothing is emitted. */
+  private readonly reflectEffortHook: HookCallback = (input) => {
+    if (this.effort === undefined && input.effort?.level) {
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply packages/agent-adapter/src/native/claude-code.ts:315]: This is a false positive. `effort?: { level: string }` is defined on `BaseHookInput` (sdk.d.ts ~172-176), and `StopHookInput = BaseHookInput & {…}`, so every `HookInput` member carries it — `input.effort?.level` typechecks. The package's `tsc --noEmit` is green on this path. The `(input: unknown)` cast in the test is only to pull the callback back out of the mock's `q.options.hooks` (stored as `Record<string, unknown>`), not to hide an implementation type error.
